theano/gpuarray/__init__.py

```diff
@@ -49,21 +49,25 @@ def init_dev(dev, name=None, preallocate=None):
     if (pygpu.version.major != 0 or pygpu.version.minor != 7 or
             pygpu.version.patch < 0):
         raise ValueError(
-            "Your installed version of pygpu is too old, please upgrade to 0.6.1 or later")
+            "Your installed version of pygpu is too old, please upgrade to 0.7.0 or later")
     # This is for the C headers API, we need to match the exact version.
     if pygpu.gpuarray.api_version()[0] != 2:
         raise ValueError(
             "Your installed libgpuarray is not in sync, please make sure to have the appropriate version")
     if dev not in init_dev.devmap:
+        args = dict()
         if config.gpuarray.cache_path != '':
-            os.environ['GPUARRAY_CACHE_PATH'] = config.gpuarray.cache_path
+            args['kernel_cache_path'] = config.gpuarray.cache_path
         if preallocate is None:
             preallocate = config.gpuarray.preallocate
+        if preallocate < 0:
+            args['max_cache_size'] = 0
+        else:
+            args['initial_cache_size'] = preallocate
```

Comment:
Something don't work or I don't understand. If you don't define max_cache_size, it is by default 0, but you always call gpucontext_props_alloc_cache() in the c code and that method assert that the the max size is bigger then the initial. Which end up being false here.